src/plugins/discover/public/application/main/components/field_stats_table/field_stats_tab.tsx

```diff
@@ -20,8 +25,34 @@ export const FieldStatisticsTab: React.FC<Omit<FieldStatisticsTableProps, 'query
       data: services.data,
     });
     const additionalFieldGroups = useAdditionalFieldGroups();
+    const isEsql = useIsEsqlMode();
+    const { euiTheme } = useEuiTheme();
 
     if (!services.dataVisualizer) return null;
+
+    if (isEsql) {
```

Comment:
We now disabled the button to switch to field stats in ES|QL mode, but this will be a safeguard in case user has bookmarked or restore url with the setting.